Stack/Opc.Ua.Bindings.Https/Stack/Https/HttpsTransportListener.cs

```diff
@@ -368,8 +373,28 @@ public async Task SendAsync(HttpContext context)
                     return;
                 }
 
+                string path = context.Request.Path.Value?.TrimEnd('/') ?? string.Empty;
+                
+                bool isDiscoveryPath = m_discovery.EndsWith(path, StringComparison.OrdinalIgnoreCase);
+                bool validateClientTlsCertificate = !isDiscoveryPath && m_mutualTlsEnabled;
+
                 IServiceRequest input = (IServiceRequest)BinaryDecoder.DecodeMessage(buffer, null, m_quotas.MessageContext);
 
+                // Match tls client certificate against client application certificate provided in CreateSessionRequest
+                if (validateClientTlsCertificate &&
+                    input.TypeId == DataTypeIds.CreateSessionRequest)
+                {
+                    byte[] tlsClientCertificate = context.Connection.ClientCertificate.RawData;
+                    byte[] opcUaClientCertificate = ((CreateSessionRequest)input).ClientCertificate;
+                    if (!Utils.IsEqual(tlsClientCertificate, opcUaClientCertificate))
```

Comment:
is there a check necessary if bothe certs are null, it is a not allowed condition.

Reply:
changed